hail_search/queries/both_sample_types_mixin.py

```diff
@@ -0,0 +1,141 @@
+        any_valid_entry, is_any_affected = self._get_any_family_member_gt_has_alt_filter()
+        ht = ht.annotate(
+            wes_passes=ht.wes_family_entries.map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ),
+            wgs_passes=ht.wgs_family_entries.map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            )
+        )
+
+        comp_het_ht = None
+        if self._has_comp_het_search:
+            comp_het_ht = self._annotate_families_inheritance(
+                ht, COMPOUND_HET, inheritance_filter, sorted_wes_family_sample_data,
+                sample_type='WES',
+            )
+            comp_het_ht = self._annotate_families_inheritance(
+                comp_het_ht, COMPOUND_HET, inheritance_filter, sorted_wgs_family_sample_data,
+                sample_type='WGS',
+            )
+            comp_het_ht = self._post_process_inheritance_both_sample_types(comp_het_ht)
+            comp_het_ht = comp_het_ht.filter(
+                comp_het_ht.wes_passes.any(hl.is_defined) | comp_het_ht.wgs_passes.any(hl.is_defined)
+            ).select_globals('family_guids', 'family_guids_1')
+
+        if is_any_affected or not (inheritance_filter or self._inheritance_mode):
+            # No sample-specific inheritance filtering needed
+            sorted_wes_family_sample_data = sorted_wgs_family_sample_data = []
+
+        if self._inheritance_mode == COMPOUND_HET:
+            ht = None
+        else:
+            ht = self._annotate_families_inheritance(
+                ht, self._inheritance_mode, inheritance_filter, sorted_wes_family_sample_data,
+                sample_type='WES',
+            )
+            ht = self._annotate_families_inheritance(
+                ht, self._inheritance_mode, inheritance_filter, sorted_wgs_family_sample_data,
+                sample_type='WGS',
+            )
+            ht = self._post_process_inheritance_both_sample_types(ht)
+            ht = ht.filter(
+                ht.wes_passes.any(hl.is_defined) | ht.wgs_passes.any(hl.is_defined)
+            ).select_globals('family_guids', 'family_guids_1')
+
+        return ht, comp_het_ht
+
+    def _post_process_inheritance_both_sample_types(self, ht: hl.Table):
+        if self._inheritance_mode == DE_NOVO:
+            def _validate_de_novo(sample_type_1_passes, sample_type_2_passes):
+                # Remove a variant if it was filtered out due to inheritance in one sample type
+                # and is de novo in the other sample type but there is only one affected sample.
+                return hl.enumerate(sample_type_1_passes).map(
+                    lambda family_entries: hl.if_else(  # family_entries: (idx, [<family_1_sample_1>, <family_1_sample_2>, ...]) or (idx, NA)
+                        hl.is_missing(sample_type_2_passes[family_entries[0]]) & hl.is_defined(family_entries[1]),
```

Comment:
so we can't assume that the entries arrays are properly sorted for both sample types such that the index of a family in one is the same as the index of a family in the other (i.e WES might be `['F1', 'F3', 'F5']` and WGS might be `['F2', 'F3']`, and theres no reason to assume that the entries within a family are the same as there could be different numbers of samples per sample type. I think you either need to do the normalizing/ merging logic before the filtering, or you need to be using the sample index data to explicitly compare the correct samples

Reply:
This makes sense. If I don't need to handle this special case, I don't need to keep track of sample indices.